utils/data_helpers.py

```diff
@@ -137,11 +140,21 @@ def download_flowers():
     image_labels_download_loc = pathlib.Path('data/flowers/imagelabels.mat')
     urllib.request.urlretrieve(IMAGE_LABELS_URL, image_labels_download_loc)
 
-def get_string_images(image_paths):
+def get_string_images(image_paths, image_dims):
     """ Generate a list of string representations of each image """
     str_images_list = []
     for image_path in image_paths:
-        image_string = open(image_path, 'rb').read()
+        image = Image.open(image_path, 'r')
+        old_size = image.size[:2]
+        ratio = max(image_dims)/max(old_size)
+        new_size = tuple([int(x*ratio) for x in old_size])
+        image = image.resize(new_size, Image.ANTIALIAS)
+        new_img = Image.new('RGB', image_dims)
+        new_img.paste(image, ((image_dims[0]-new_size[0])//2,
+                              (image_dims[1]-new_size[1])//2))
+        img_buffer = io.BytesIO()
+        new_img.save(img_buffer, format='PNG')
+        image_string = img_buffer.getvalue()
```

Comment:
I assume this is the code that maintains the aspect ratio?

Reply:
yup, pretty much:
1) determine aspect ratio
2) determine reduced image size required to maintain ar
3) create blank square image
4) paste resized image in middle of blank image

Reply:
:+1: